src/qibo/validation/gate_set_tomography.py

```diff
@@ -0,0 +1,2089 @@
+def sort_counts_1qb(matrix):
+    import numpy as np
+
+    counts_matrix = np.zeros((2, 2))
+
+    for ii in range(0, 2):
+        word = bin(ii)[2:]
+        counts_matrix[ii, 0] = word
+
+    for row in matrix:
+        row_number = int(str(int(row[0])), 2)
+        counts_matrix[row_number, 1] = row[1]
+
+    return counts_matrix
+
+
+def sort_counts_2qb(matrix):
+    import numpy as np
+
+    counts_matrix = np.zeros((4, 2))
+
+    for ii in range(0, 4):
+        word = bin(ii)[2:]
+        counts_matrix[ii, 0] = word
+
+    for row in matrix:
+        row_number = int(str(int(row[0])), 2)
+        counts_matrix[row_number, 1] = row[1]
+
+    return counts_matrix
+
+
+def GST_1qb(
+    NshotsGST=1e4,
+    noise_model=None,
+    save_data=None,
+    class_qubit_gate=None,
+    theta=None,
+    operator_name=None,
+    backend=None,
+):
+    """Runs gate set tomography for a single qubit gate/circuit.
+
+        Args:
+        NshotsGST (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied
+            to simulate noisy computation.
+        save_data: To save gate set tomography data. If None, skip.
+        class_qubit_gate: Identifier for the type of gate given in qibo.raw. If None,
+            do gate set tomography with empty circuit.
+        theta: Identifier for the gate parameter given in qibo.raw. If None, do gate
+            set tomography with only gate provided by class_qubit_gate.
+        operator_name: Name of gate given in qibo.raw used for saving data.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+
+    Returns:
+        numpy.matrix: Matrix with elements jk equivalent to either Tr(Q_j rho_k) or Tr(Q_j O_l rho_k).
+    """
+
+    import numpy as np
+
+    from qibo import gates
+    from qibo.backends import GlobalBackend
+    from qibo.models import Circuit
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    if class_qubit_gate is None:
+        print("Empty circuit without any operator (1 qb)")
```

Comment:
Nope it's not supposed to flag an error. So GST_1qb and GST_2qb has an optional parameter called class_qubit_gate. If class_qubit_gate is present, then gate set tomography is performed for that input (i.e. the O_jk in my slide that I sent). If it's not present, then gate set tomography is performed for an empty circuit (i.e. the g_jk matrix in the slides). 

I just set it to print something so that the user knows that there is / isn't any gate given to the function. Do you think it's necessary?

Reply:
No, if not having gates is a problem I would throw an error, otherwise I'll remove it. This print is also one of the reasons the tests are failing. In qibo functions I don't think you are supposed to print things.

Reply:
Got it! I will leave the print statements when checking the code locally but remove the print statements when I push it back.